libs/ui/src/screens/send-confirmation/components/approve-token/approve-token.tsx

```diff
@@ -59,7 +59,8 @@ export const ApproveToken: FC<Props> = ({
   const { explorerUrl } = useSelectedNetworkSelector();
   const gasToken = useGasTokenSelector();
 
-  const { spender, allowanceAmount } = useMemo(() => getDecodedApproveTokenData(data as string), [data]);
+  const { spender, allowanceAmount } = useMemo(() => getDecodedApproveTokenData(data), [data]);
+  const proposedAllowanceAmount = useMemo(() => allowanceAmount, []);
```

Comment:
it is used instead of `useRef`?

Reply:
yes
we just need to store the first value that dapp or our swap provided because then we can change the allowance amount